euphonic/qpoint_frequencies.py

```diff
@@ -168,6 +188,17 @@ def _calculate_dos(self, dos_bins: Quantity,
                     pdf = norm.pdf(dos_bins_calc, loc=freqs[q,m],
                                    scale=mode_widths[q,m])
                     dos += pdf*weights[q]*mode_weights_calc[q, m]
+        elif mode_widths is not None and adaptive_method=='fast':
+            # fast, approximate method for adaptive broadening
+            freqs = np.ravel(freqs)
+            mode_widths = np.ravel(mode_widths).to('hartree').magnitude
+            n_weights = np.repeat(weights, n_modes)
+
+            # set default error value if none specified
+            if error is None:
+                error = 0.01
```

Comment:
I don't see the harm in this having a default value in the function signature.

Admittedly we are accumulating arguments that "don't always do anything". `mode_widths_min` was already one of those. But short of breaking some argument groups into dicts (which impacts usability) I don't see a nice way of tidying it up 🤔

Reply:
The reason I'd set the default value to None in the function signature was to avoid the case (as noted with the adaptive_method warning) where the user isn't warned if they happen to non-sensically set the error to the default option - I'll have a think about whether this approach can be modified/improved

Reply:
If it is also named `adaptive_error` the problem is reduced a bit. (Or at least we can lay more blame on the user.)

If we are going to attach warnings to over half the arguments on this function, it should be done in a systematic way. But it has "bad smell" either way - is there a better way to avoid this problem?

Reply:
I agree `adaptive_error` should be set in the function signature. I don't think we can/should give warnings for all sets of nonsensical arguments, as it could make the code more bloated and confusing if anything. For example, there's no warning elsewhere in Euphonic if someone sets `use_c=False` but `n_threads=4`. If we make the connections between arguments obvious enough in the docstring/variable names it should be ok, we should give the user _some_ credit. 

To give one other example, I've just had a look at [`numpy.histogram`](https://numpy.org/doc/stable/reference/generated/numpy.histogram.html) and it has a similar issue with having to support many potential options. If you give it an array of bin edges as the `bins` argument, the `range` argument is then ignored. Although you might suspect this it isn't stated in the docstring though.

Anyway, back to looking at this PR...